frontend/app/src/people/widgetViews/orgnization/WithdrawBudgetModal.tsx

```diff
@@ -0,0 +1,233 @@
+  width: 56px;
+  height: 56px;
+  margin-bottom: 30px;
+`;
+
+const SuccessText = styled.p`
+  color: #8e969c;
+  font-size: 0.95rem;
+  font-weight: 500;
+  margin-top: 10px;
+`;
+
+const ErrorText = styled(SuccessText)`
+  color: #ed7474;
+`;
+
+const WithdrawBudgetModal = (props: WithdrawModalProps) => {
+  const [paymentRequest, setPaymentRequest] = useState('');
+  const [amountInSats, setAmountInSats] = useState(0);
+  const [paymentSettled, setPaymentSettled] = useState(false);
+  const [paymentError, setPaymentError] = useState('');
+
+  const isMobile = useIsMobile();
+  const { ui, main } = useStores();
+  const { isOpen, close, uuid, getOrganizationBudget } = props;
+
+  function isBudgetSuccess(object: any): object is BudgetWithdrawSuccess {
+    return 'response' in object;
+  }
+
+  const getAndSetDetails = async (paymentRequest: string) => {
+    try {
+      const decoded = LighningDecoder.decode(paymentRequest);
+      const sats = decoded.sections[2].value / 1000;
+      setAmountInSats(sats);
+    } catch (e) {
+      console.log(`Cannot decode lightning invoice: ${e}`);
+    }
+  };
+
+  const withdrawBudget = async () => {
+    const token = ui.meInfo?.websocketToken;
+    const body = {
+      org_uuid: uuid ?? '',
+      payment_request: paymentRequest,
+      websocket_token: token
+    };
+
+    const response = await main.withdrawBountyBudget(body);
+    if (isBudgetSuccess(response)) {
+      await getOrganizationBudget();
+      setPaymentSettled(true);
+    } else {
+      setPaymentError(response.error);
+    }
+  };
+
+  const getInvoiceDetails = async (paymentRequest: string) => {
+    getAndSetDetails(paymentRequest);
+  };
+
+  return (
+    <Modal
+      visible={isOpen}
+      style={{
+        height: '100%',
+        flexDirection: 'column',
+        padding: '10px'
+      }}
+      envStyle={{
+        marginTop: isMobile ? 64 : 0,
+        background: color.pureWhite,
+        zIndex: 20
+      }}
+      overlayClick={close}
+      bigCloseImage={close}
+      bigCloseImageStyle={{
+        top: '-18px',
+        right: '-18px',
+        background: '#000',
+        borderRadius: '50%'
+      }}
+    >
+      <Wrap newDesign={true}>
+        {paymentSettled && (
+          <PaymentDetailsWrap>
+            <SuccessImage src={successIcon} />
+            <WithdrawAmount>
+              {amountInSats.toLocaleString()} <Grey>SATS</Grey>
+            </WithdrawAmount>
+            <SuccessText>Successfully Withdraw</SuccessText>
+          </PaymentDetailsWrap>
+        )}
+        {paymentError && (
+          <PaymentDetailsWrap>
+            <SuccessImage src={errorIcon} />
+            <WithdrawAmount>
+              {amountInSats.toLocaleString()} <Grey>SATS</Grey>
+            </WithdrawAmount>
+            <ErrorText>{paymentError}</ErrorText>
+          </PaymentDetailsWrap>
+        )}
+        {amountInSats > 0 && !paymentSettled && !paymentError && ui.meInfo?.owner_pubkey && (
```

Comment:
lets make this a boolean defined above it looks a bit messy to make a statment like this
`const showPaymentDetails = amountInSats > 0 && !paymentSettled && !paymentError && ui.meInfo?.owner_pubkey`